bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/eGateHandler.java

```diff
@@ -0,0 +1,344 @@
+        }
+    }
+
+    public void pulseUp(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=1;Priority=1;CheckNr=3415347;" + CR);
+    }
+
+    public void pulseDown(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=2;Priority=1;CheckNr=2764516;" + CR);
+    }
+
+    public void continuousUp(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=3;Priority=1;CheckNr=2867016;" + CR, 20000);
+    }
+
+    public void continuousDown(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=4;Priority=1;CheckNr=973898;" + CR, 20000);
+    }
+
+    public void stop(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=5;Priority=1;CheckNr=5408219;" + CR);
+    }
+
+    @SuppressWarnings("null")
+    public void registerBlind(String id, ThingUID uid) {
+        logger.debug("Registring Blind id {} with thingUID {}", id, uid);
+        registeredBlinds.put(id, uid);
+    }
+
+    /**
+     * Send a command to the eGate Server.
+     */
+
+    private void sendCommand(String command) {
+        sendCommand(command, SOCKET_TIMEOUT);
+    }
+
+    @SuppressWarnings("null")
+    private synchronized void sendCommand(String command, int timeout) {
+        logger.debug("EGate got command: {}", command);
+
+        if (!isConnected()) {
+            logger.debug("no connection to Dominoswiss eGate server when trying to send command, returning...");
+            return;
+        }
+
+        // Send plain string to eGate Server,
+        logger.debug("Sending command: {}", command);
+        try {
+
+            // BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(egateSocket.getOutputStream()));
+
+            egateSocket.setSoTimeout(SOCKET_TIMEOUT);
+            writer.write(command);
+            writer.flush();
+            logger.debug("Sent message");
+        } catch (IOException e) {
+            logger.error("Error while sending command {} to Dominoswiss eGate Server {} ", command, e.toString());
+        }
+    }
+
+    private void startAutomaticRefresh() {
+        Runnable runnable = new Runnable() {
+            @SuppressWarnings("null")
+            @Override
+            public void run() {
+                try {
+                    if (reader == null) {
+                        reader = new BufferedReader(new InputStreamReader(egateSocket.getInputStream()));
+                    }
+                    logger.debug("Socket State: {} to: {}", egateSocket.isConnected(), egateSocket.toString());
+                    if (reader.ready()) {
+                        String input = reader.readLine();
+                        logger.debug("Reader got from EGATE: {}", input);
+                        onData(input);
+                    }
+                } catch (IOException e) {
+                    logger.error("Error while reading command from Dominoswiss eGate Server {} ", e.toString());
+                    updateStatus(ThingStatus.OFFLINE);
```

Comment:
You could add the exception's message as a third argument to `updateStatus()` to display it in the UI.